packages/copyable/src/Copyable/Copyable.tsx

```diff
@@ -85,6 +86,10 @@ export default function Copyable({
       container: portalContainer,
     });
 
+    clipboard.on('success', (event: React.ClipboardEvent<HTMLDivElement>) => {
+      onCopy?.(event);
+    });
```

Comment:
![Screenshot 2024-06-12 at 1 50 25 PM](https://github.com/mongodb/leafygreen-ui/assets/52300529/8494af32-a354-466a-b6cc-97823933e8ef)
The ticket says I can't view but was looking to verify if we only want to fire this on `'success'`. `onCopy` suggests this callback should run any time a user attempts to copy (e.g. if a product team added analytics to capture button clicks)

If the ticket specifies we only want to fire this on `'success'`, can we call the prop `onCopySuccess`? That said, I think an `onCopy` prop makes sense to introduce as well. It would be good to retain parity with the implementation of the `CopyButton` component in the `Code` package
https://github.com/mongodb/leafygreen-ui/blob/9afcc3bdbfe966ae894903fdeb990b3a3c799ac9/packages/code/src/CopyButton/CopyButton.tsx#L65-L69

Reply:
Added on fail too - good call